### DISCUSSION TITLE
The point of the default.crt?

### DISCUSSION THREAD

**Discussion by hardillb** [nginx-proxy/nginx-proxy#2501] [Q&A] [answered]
The docs contain the following:

> ## Missing Certificate
> 
> If no matching certificate is found for a given virtual host, nginx-proxy will:
> 
>   - configure nginx to use the default certificate (default.crt with default.key) and return a 500 error for HTTPS,
>   - force enable HTTP; i.e. HTTPS_METHOD will switch to noredirect if it was set to nohttp or redirect. If this switch to HTTP is not wanted set ENABLE_HTTP_ON_MISSING_CERT=false (default is true).

Can I ask what the point of setting the default certificate is if it is always going to return a 500 error?

Could we have an option to just reply with the default.crt and not return the error?
Also has this behaviour changed? I'm sure it used to just return the default.crt.

In my case the default.crt actually matched the hostname in question so would have worked and there was nothing in the logs explaining why the 500 error was returned, it was only when I dumped the /etc/nginx/default.conf file that I discovered the 

```
if ($https) {
    return 500;
}

```

**Comment by buchdag** [comment]
Hi @hardillb, sorry for the late answer.

Thank you for pointing at this, I agree that this should not be the way `default.crt` works. I'll look into it.

> Also has this behaviour changed? I'm sure it used to just return the default.crt.

It did, we had trouble with serving incorrect hosts on some cases in the past and if I remember correctly this was done in an effort to fix this. The template code was also heavily refactored during this period and I missed the fact that we broke an expected behaviour.

**Reply by buchdag** [reply, marked as answer]
This will be fixed in #2535
